src/map/clif.cpp

```diff
@@ -25550,6 +25550,108 @@ void clif_parse_macro_checker( int32 fd, map_session_data* sd ){
 #endif
 }
 
+/*==========================================
+ * macro user report client packet processing function
+ *------------------------------------------*/
+void clif_parse_macro_user_report(int32 fd, map_session_data *sd)
+{
+#if (PACKETVER_MAIN_NUM > 20230915)
+	nullpo_retv(sd);
+
+	PACKET_CZ_MACRO_USER_REPORT_REQ* const Packet = reinterpret_cast<PACKET_CZ_MACRO_USER_REPORT_REQ*>(RFIFOP(fd, 0));
+
+	//
+	// Packets that may be forged needs to be integrity checked before processing them.
+	//
+
+	if (Packet->ReportType > 1)
+	{
+		return;
+	}
+
+	if (Packet->ReporterAID == Packet->ReportedAID)
+	{
+		return;
+	}
+
+	if (Packet->ReporterAID != sd->status.account_id)
+	{
+		return;
+	}
+
+	map_session_data* tsd = map_id2sd(Packet->ReportedAID);
+	if (tsd == nullptr)
+	{
+		clif_macro_user_report_ack(sd, MACRO_USER_REPORT_INVALID, nullptr);
+		return;
+	}
+
+	//
+	// Checks whether the reported user character name matches.
+	//
+
+	if (strcmpi(Packet->ReportName, tsd->status.name) != 0)
+	{
+		clif_macro_user_report_ack(sd, MACRO_USER_REPORT_INVALID, nullptr);
+		return;
+	}
+
+	//
+	// Limits the maximum report count per reporter user.
+	//
+
+	const uint32 ReportCount = static_cast<uint32>(pc_readreg2(sd, "#MUR_ReportCount"));
+	if (ReportCount > 999)
+	{
+		clif_macro_user_report_ack(sd, MACRO_USER_REPORT_COUNTLIMIT, nullptr);
+		return;
+	}
+
+	pc_setreg2(sd, "#MUR_ReportCount", ReportCount + 1);
```

Comment:
```suggestion
	pc_setreg2(sd, "#MUR_ReportCount", reportCount + 1);
```
And please use the define from pc.hpp that you will create